src/types.rs

```diff
@@ -600,14 +610,14 @@ pub struct ContractAddress {
     serde(tag = "type", content = "address", rename_all = "lowercase")
 )]
 #[cfg_attr(feature = "fuzz", derive(Arbitrary))]
-#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+#[derive(PartialEq, Eq, Copy, Clone, PartialOrd, Ord, Debug, Hash)]
```

Comment:
Did you check what the generated comparison function is and whether we can make one that generates smaller code?